go/libraries/doltcore/merge/schema_merge_test.go

```diff
@@ -1354,10 +1369,246 @@ var jsonMergeTests = []schemaMergeTest{
+				{
+					name:     `convergent deletion`,
+					ancestor: singleRow(1, 1, 1, insert(insert(largeObject, "$.a", 1), "$.z", 2)),
+					left:     singleRow(1, 2, 1, insert(largeObject, "$.a", 1)),
+					right:    singleRow(1, 1, 2, insert(largeObject, "$.z", 2)),
+					merged:   singleRow(1, 2, 2, largeObject),
+				},
+				{
+					name:         `divergent insertion`,
+					ancestor:     singleRow(1, 1, 1, largeObject),
+					left:         singleRow(1, 2, 1, insert(largeObject, "$.z", 1)),
+					right:        singleRow(1, 1, 2, insert(largeObject, "$.z", 2)),
+					dataConflict: true,
+				},
+				{
+					name:         `divergent modification`,
+					ancestor:     singleRow(1, 1, 1, largeObject),
+					left:         singleRow(1, 2, 1, set(largeObject, "$.level7", 1)),
+					right:        singleRow(1, 1, 2, set(largeObject, "$.level7", 2)),
+					dataConflict: true,
+				},
+				{
+					name:         `divergent modification and deletion`,
+					ancestor:     singleRow(1, 1, 1, insert(largeObject, "$.a", 1)),
+					left:         singleRow(1, 2, 1, insert(largeObject, "$.a", 2)),
+					right:        singleRow(1, 1, 2, largeObject),
+					dataConflict: true,
+				},
+				{
+					name:     `nested insertion`,
+					ancestor: singleRow(1, 1, 1, insert(largeObject, "$.level7.level4.new", map[string]interface{}{})),
+					left:     singleRow(1, 2, 1, insert(largeObject, "$.level7.level4.new", map[string]interface{}{"a": 1})),
+					right:    singleRow(1, 1, 2, insert(largeObject, "$.level7.level4.new", map[string]interface{}{"b": 2})),
+					merged:   singleRow(1, 2, 2, insert(largeObject, "$.level7.level4.new", map[string]interface{}{"a": 1, "b": 2})),
+				},
+				{
+					name:     `nested insertion with escaped quotes in keys`,
+					ancestor: singleRow(1, 1, 1, insert(largeObject, `$.level7.level4."\"new\""`, map[string]interface{}{})),
+					left:     singleRow(1, 2, 1, insert(largeObject, `$.level7.level4."\"new\""`, map[string]interface{}{"a": 1})),
+					right:    singleRow(1, 1, 2, insert(largeObject, `$.level7.level4."\"new\""`, map[string]interface{}{"b": 2})),
+					merged:   singleRow(1, 2, 2, insert(largeObject, `$.level7.level4."\"new\""`, map[string]interface{}{"a": 1, "b": 2})),
+				},
+				{
+					name:     "nested convergent modification",
+					ancestor: singleRow(1, 1, 1, largeObject),
+					left:     singleRow(1, 2, 1, set(largeObject, "$.level7.level4", 1)),
+					right:    singleRow(1, 1, 2, set(largeObject, "$.level7.level5", 2)),
+					merged:   singleRow(1, 2, 2, set(set(largeObject, "$.level7.level5", 2), "$.level7.level4", 1)),
+					// skip:     true, // panics
+				},
+				{
+					name:     `nested deletion`,
+					ancestor: singleRow(1, 1, 1, largeObject),
+					left:     singleRow(1, 2, 1, delete(largeObject, "$.level7")),
+					right:    singleRow(1, 1, 2, delete(largeObject, "$.level6")),
+					merged:   singleRow(1, 2, 2, delete(delete(largeObject, "$.level6"), "$.level7")),
+					skip:     true, // doesn't terminate
```

Comment:
Sounds like this gets the engine wedged? If so, we should fix this before we merge this in so that we don't let customers hit this.

Reply:
Thanks for catching. The test works, I just forgot to re-enable it.